daemon/daemon_main.go

```diff
@@ -1114,6 +1115,11 @@ func runDaemon() {
 		d.workloadsEventsCh = eventsCh
 	}
 
+	if k8s.IsEnabled() {
```

Comment:
What is the need for this change? Please explain why this is necessary in the commit message. I am wary of changing orders of operations upon daemon bootstrap.